R/particle_filter.R

```diff
@@ -152,6 +160,7 @@ particle_filter <- R6::R6Class(
 
       self$history <- history
       self$state <- state
+      private$last_model <- model
```

Comment:
Why is this used rather than `self$model`? Is it possible to modify this outside of the object? If so, why not make it private and add an access function? (Or is that too based upon how I'm used to using objects in C++?)

Reply:
The `$model` field is the generator, `$last_model` is the model itself with parameters set in it. I'd hoped to use `$set_user()` but was getting caught between two poor behaviours in odin's handling of the ring buffer. This way is somewhat easier to think about at laest